tests/upgrades/test_virtwho.py

```diff
@@ -0,0 +1,194 @@
+"""Test for Virt-who related Upgrade Scenario's
+
+:Requirement: Upgraded Satellite
+
+:CaseAutomation: Automated
+
+:CaseLevel: Acceptance
+
+:CaseComponent: API
+
+:TestType: Functional
+
+:CaseImportance: High
+
+:Upstream: No
+"""
+
+from fauxfactory import gen_string
+from nailgun import entities
+from robottelo.decorators import skip_if_not_set
+from robottelo.test import APITestCase, settings
+from robottelo.virtwho_utils import (
+    deploy_configure_by_command,
+    get_configure_command,
+    get_configure_file,
+    get_configure_option,
+    get_hypervisor_info
+)
+from upgrade_tests import pre_upgrade, post_upgrade
+from wait_for import wait_for
+
+
+class scenario_positive_virt_who(APITestCase):
+    """Virt-who config is intact post upgrade and verify virt-who update/delete
+
+    :steps:
+
+        1. In Preupgrade Satellite, create virt-who-config
+        2. Upgrade the satellite to next/latest version
+        3. Postupgrade, Verify the virt-who config is intact, update and delete
+
+    :expectedresults: Virtwho config should be created, updated and deleted successfully.
+    """
+    @classmethod
+    @skip_if_not_set('virtwho')
+    def setUpClass(cls):
+        cls.org = entities.Organization().search(
+            query={'search': 'name="Default Organization"'}
+        )[0]
+        cls.satellite_url = settings.server.hostname
+        cls.hypervisor_type = settings.virtwho.hypervisor_type
+        cls.hypervisor_server = settings.virtwho.hypervisor_server
+        cls.hypervisor_username = settings.virtwho.hypervisor_username
+        cls.hypervisor_password = settings.virtwho.hypervisor_password
+        cls.hypervisor_config_file = settings.virtwho.hypervisor_config_file
+        cls.vdc_physical = settings.virtwho.sku_vdc_physical
+        cls.vdc_virtual = settings.virtwho.sku_vdc_virtual
+
+        cls.name = 'preupgrade_virt_who'
+
+    def _make_virtwho_configure(self):
+        args = {
+            u'debug': 1,
+            u'interval': '60',
+            u'hypervisor_id': 'hostname',
+            u'hypervisor_type': self.hypervisor_type,
+            u'hypervisor_server': self.hypervisor_server,
+            u'organization_id': self.org.id,
+            u'filtering_mode': 'none',
+            u'satellite_url': self.satellite_url,
+        }
+        if self.hypervisor_type == 'libvirt':
+            args[u'hypervisor_username'] = self.hypervisor_username
+        elif self.hypervisor_type == 'kubevirt':
+            args[u'kubeconfig'] = self.hypervisor_config_file
+        else:
+            args[u'hypervisor_username'] = self.hypervisor_username
+            args[u'hypervisor_password'] = self.hypervisor_password
+        return args
+
+    def _try_to_get_guest_bonus(self, hypervisor_name, sku):
+        subscriptions = entities.Subscription().search(
+            query={'search': '{0}'.format(sku)})
```

Comment:
One more instance of needless `format`. Did not notice it before:
```suggestion
            query={'search': sku})
```

Reply:
oh, thanks